department-of-veterans-affairs/caseflow
# APPEALS-36720 | Add Error Handling to Frontend for Report Generation

Jira: https://jira.devops.va.gov/browse/APPEALS-37620

# Description
Add in frontend error handling for if the report generation request fails.
Also add tests to force a failure and check for the error message.


## Acceptance Criteria
- [ ] Error message is shown when request fails
- [ ] The screen focuses on the error message on initial showing

## Testing Plan
1. Log in as VHAADMIN1
2. Go to `/decision_reviews/vha/report`
3. Get the form ready for submission
4. Go to the chrome dev tools and under "Network" click "No throttling" and set it to "Offline" as seen below
(This will force an error. Make sure to reset this back to "No throttling" after you are done testing)
5. Click "Generate Task Report
6. Test should fail
 
![network](https://github.com/department-of-veterans-affairs/caseflow/assets/51007432/ba802e56-91ee-4362-9790-c78b8061b2c9)


# Screenshot
![Screenshot 2024-01-08 at 3 53 58 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/51007432/dbe2b099-327d-496a-8255-50e2ec597c04)
